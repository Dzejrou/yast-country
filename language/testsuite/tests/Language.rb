# encoding: utf-8

# tests for Language.ycp functions
# Maintainer:	jsuchome@suse.cz
# $Id$
module Yast
  class LanguageClient < Client
    def main
      # testedfiles: Language.ycp

      Yast.include self, "testsuite.rb"

      @READ = {
        "sysconfig" => { "language" => { "DEFAULT_LANGUAGE" => "en_US" } },
        "target"    => { "bash_output" => {}, "size" => 1, "yast2" => {} }
      }

      TESTSUITE_INIT([@READ, {}, @READ], nil)
      Yast.import "Pkg"
      Yast.import "Language"

      DUMP("GetLanguageCountry")
      Language.language = "de_AT@UTF-8"
      TEST(lambda { Language.GetLanguageCountry }, [], nil)
      Language.language = "de_AT"
      TEST(lambda { Language.GetLanguageCountry }, [], nil)
      Language.language = "de"
      TEST(lambda { Language.GetLanguageCountry }, [], nil)

      Language.language = "" # use default language

      TEST(lambda { Language.GetLanguageCountry }, [], nil)

      nil
    end
  end
end

Yast::LanguageClient.new.main
