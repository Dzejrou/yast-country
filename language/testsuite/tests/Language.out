Read	.target.yast2 "lang2iso.ycp" nil
Read	.target.size "/etc/sysconfig/language" -1
Dump	GetLanguageCountry
Return	AT
Return	AT
Return	DE
Return	
